frontend/src/views/SettingsView.vue

```diff
@@ -1,7 +1,110 @@
 <template>
 	<div class="page_box_wrapper">
 		<div class="page_box">
-			<h1>This is a profile settings page</h1>
+			<h1>SETTINGS</h1>
+			<ProfileSettingAvatar :profile_picture="me.avatar" />
+			<div class="c_block c_form_group tac">
+				<ChangeAvatar />
+			</div>
+			<form
+				method="Post"
+				action=""
+				class="c_block c_form_group"
+				@submit.prevent="updateProfile(me.id, updateProfileForm)"
+			>
+				<InputField
+					v-if="me.is_intra"
+					label="intra name"
+					v-model="me.intra_login"
+					:is_disabled="true"
+				/>
+				<InputField
+					label="display_name"
+					:modelValue="me.name"
+					v-model="updateProfileForm.name"
+				/>
+				<InputField
+					v-if="!me.is_intra"
+					label="new password"
+					v-model="updateProfileForm.new_password"
+				/>
+				<InputField
+					v-if="!me.is_intra"
+					label="new password confirm"
+					v-model="updateProfileForm.new_password_confirm"
+				/>
+				<InputField
+					v-if="!me.is_intra"
+					label="current password"
+					v-model="updateProfileForm.password"
+				/>
+				<div class="c_block c_split">
+					<p><a href="/turnon2fa">turn on 2fa</a></p>
+				</div>
+				<div class="page_button pb_bottom">
+					<input type="submit" value="save" />
+				</div>
+				<div v-if="errors.length">
+					<p v-for="error in errors" class="c_form--error">
+						!! {{ error }}
+					</p>
+				</div>
+			</form>
 		</div>
 	</div>
 </template>
+
+<script lang="ts">
+import InputField from '@/components/input-fields/InputField.vue';
+import { useUserStore } from '@/stores/userStore';
+import { defineComponent, reactive } from 'vue';
+import type { UpdateProfileForm } from '@/types/User';
+import ProfileList from '@/views/ProfilesView.vue';
+import ProfileSettingAvatar from '@/components/user-info/ProfileSettingAvatar.vue';
+import ChangeAvatar from '@/components/input-fields/ChangeAvatar.vue';
+import { storeToRefs } from 'pinia';
+
+export default defineComponent({
+	name: 'RegisterView',
+	components: {
+		ChangeAvatar,
+		ProfileSettingAvatar,
+		ProfileList,
+		InputField,
+	},
+
+	async created() {
+		await useUserStore().refreshMe();
+	},
+
+	setup() {
+		const userStore = useUserStore();
+		useUserStore().refreshMe();
+		const { me, errors } = storeToRefs(userStore);
+		const { updateProfile, refreshData } = userStore;
+		let updateProfileForm: UpdateProfileForm = reactive({
+			name: me.value.name,
+			new_password: '',
+			new_password_confirm: '',
+			password: '',
+		});
+		// if (me === undefined) {
+		// this.$router.push('/profiles');
+		// }
```

Comment:
comment can be removed

Reply:
I wanted to keep this, as a reminder, what do we do is me doesnt existsbut we'll prob handle this with guards?

Reply:
that should indeed be handled with guards at some point
if you wanna keep it, can you than add a textual part to the comment explaining that this should be implemented through a guard?
Because without that context the comment is odd

Reply:
Ill create a ticket